### DISCUSSION TITLE
is there a way to cache DNS requests for all subdomains without wildcard * ?

### DISCUSSION THREAD

**Discussion by SriHarsha001** [coredns/coredns#6133] [Q&A] [answered]
How to enable caching for all subdomains of example domain using below configuration in coredns-custom configmap ? 

example.com:53 {
  cache 300    
}
*.com:53 {
  cache 1500
}

Does it cache only DNS queries to example.com or subdomains are also cached ? 

If we use *.example.com, we get a warning (Due to this PR - https://github.com/coredns/coredns/pull/5414 and mentioned in this release notes - https://coredns.io/2022/09/07/coredns-1.9.4-release/), 

So is there any way to cache DNS requests for all subdomains of example.com ?

Going through the doc here - https://coredns.io/plugins/cache/#examples, I don't see an option to cache all subdomains (not using *). 

The only other thing I can think of is - specifying multiple subdomains (but what if we have a lot of subdomains ?)

	subdomain1.example.com {
		cache 300
	}

	subdomain2.example.com {
		cache 300
	}

Do you have any pointers or suggestions ?

**Comment by chrisohaver** [comment, marked as answer]
The following caches `example.com` and all subdomains of `example.com`.

```
example.com {
  cache 300
}
```
